// @1password/sdk/dist/items_shares.js
"use strict";
// Code generated by op-codegen - DO NOT EDIT MANUALLY
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _ItemsShares_inner;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemsShares = void 0;
const types_js_1 = require("./types.js");
class ItemsShares {
    constructor(inner) {
        _ItemsShares_inner.set(this, void 0);
        __classPrivateFieldSet(this, _ItemsShares_inner, inner, "f");
    }
    /**
     * Get the item sharing policy of your account.
     */
    getAccountPolicy(vaultId, itemId) {
        return __awaiter(this, void 0, void 0, function* () {
            const invocationConfig = {
                invocation: {
                    clientId: __classPrivateFieldGet(this, _ItemsShares_inner, "f").id,
                    parameters: {
                        name: "ItemsSharesGetAccountPolicy",
                        parameters: {
                            vault_id: vaultId,
                            item_id: itemId,
                        },
                    },
                },
            };
            return JSON.parse(yield __classPrivateFieldGet(this, _ItemsShares_inner, "f").core.invoke(invocationConfig), types_js_1.ReviverFunc);
        });
    }
    /**
     * Validate the recipients of an item sharing link.
     */
    validateRecipients(policy, recipients) {
        return __awaiter(this, void 0, void 0, function* () {
            const invocationConfig = {
                invocation: {
                    clientId: __classPrivateFieldGet(this, _ItemsShares_inner, "f").id,
                    parameters: {
                        name: "ItemsSharesValidateRecipients",
                        parameters: {
                            policy,
                            recipients,
                        },
                    },
                },
            };
            return JSON.parse(yield __classPrivateFieldGet(this, _ItemsShares_inner, "f").core.invoke(invocationConfig), types_js_1.ReviverFunc);
        });
    }
    /**
     * Create a new item sharing link.
     */
    create(item, policy, params) {
        return __awaiter(this, void 0, void 0, function* () {
            const invocationConfig = {
                invocation: {
                    clientId: __classPrivateFieldGet(this, _ItemsShares_inner, "f").id,
                    parameters: {
                        name: "ItemsSharesCreate",
                        parameters: {
                            item,
                            policy,
                            params,
                        },
                    },
                },
            };
            return JSON.parse(yield __classPrivateFieldGet(this, _ItemsShares_inner, "f").core.invoke(invocationConfig), types_js_1.ReviverFunc);
        });
    }
}
exports.ItemsShares = ItemsShares;
_ItemsShares_inner = new WeakMap();
